docs/entities/renderer.md

```diff
@@ -0,0 +1,584 @@
+    public MyEntityRenderer(EntityRendererProvider.Context context) {
+        super(context);
+        this.addLayer(new MyRenderLayer(this, context.getModelSet()));
+    }
+
+    public MyEntityRenderState createRenderState() {
+        return new MyEntityRenderState();
+    }
+
+    @Override
+    public void extractRenderState(MyEntity entity, MyEntityRenderState state, float partialTick) {
+        super.extractRenderState(entity, state, partialTick);
+        // ...
+    }
+
+    @Override
+    public void render(MyEntityRenderState state, PoseStack poseStack, MultiBufferSource bufferSource, int packedLight) {
+        super.render(state, entityYaw, partialTick, poseStack, bufferSource, packedLight);
+        // ...
+    }
+}
+```
+
+```java
+@SubscribeEvent
+public static void registerLayerDefinitions(EntityRenderersEvent.RegisterLayerDefinitions event) {
+    event.add(MyEntityModel.MY_LAYER, MyEntityModel::createBodyLayer);
+}
+
+@SubscribeEvent
+public static void registerEntityRenderers(EntityRenderersEvent.RegisterRenderers event) {
+    event.registerEntityRenderer(MY_ENTITY_TYPE.get(), MyEntityRenderer::new);
+}
+```
+
+## Modifying Existing Entity Renderers
+
+In some scenarios, it is desirable to add to an existing entity renderer, e.g. for rendering additional effects on an existing entity. Most of the time, this will affect living entities, i.e., entities with a `LivingEntityRenderer`. This enables us to add [render layers][renderlayer] to an entity like so:
+
+```java
+@SubscribeEvent
+public static void addLayers(EntityRenderersEvent.AddLayers event) {
+    // Add a layer to every single entity type.
+    for (EntityType<?> entityType : event.getEntityTypes()) {
+        // Get our renderer.
+        EntityRenderer<?, ?> renderer = event.getRenderer(entityType);
+        // Null-check the renderer. You could add more checks here,
+        // e.g., instanceof checks to only target specific renderer classes.
+        if (renderer != null) {
+            // Add the layer to the renderer. Like above, construct a new MyRenderLayer.
+            // The EntityModelSet can be retrieved from the event through #getEntityModels.
+            renderer.addLayer(new MyRenderLayer(renderer, event.getEntityModels()));
+        }
+    }
+}
+```
+
+For players, a bit of special-casing is required because there can actually be multiple player renderers. These are managed separately by the event. We can interact with them like so:
+
+```java
+@SubscribeEvent
+public static void addPlayerLayers(EntityRenderersEvent.AddLayers event) {
+    // Iterate over all possible player models.
+    for (PlayerSkin.Model skin : event.getSkins()) {
+        // Get the associated PlayerRenderer.
+        if (event.getSkin(skin) instanceof PlayerRenderer playerRenderer) {
+            // Add the layer to the renderer.
+            playerRenderer.addLayer(new MyRenderLayer(playerRenderer, event.getEntityModels()));
```

Comment:
This would not compile due to the generics differing between the layer and the renderer.